### DISCUSSION TITLE
environment variables in docker-compose

### DISCUSSION THREAD

**Discussion by christophedc0** [Koenkk/zigbee2mqtt#7807] [General]
Hi,

I'm trying to put the devices.yaml and groups.yaml file straight into the docker-compose file.

I have tried the following without success (and some variants on it), I hope someone can guide me in the right direction.
```yaml
environment:
  - ZIGBEE2MQTT_CONFIG_DEVICES=devices.yaml
  - ZIGBEE2MQTT_CONFIG_GROUPS=groups.yaml
 ```

I know it can be done in the configuration.yaml, but I'd like to put it into docker-compose as for me it's part of the basic config

**Comment by onedr0p** [comment]
I am having this issue as well, the error z2m has is:

```
Refusing to start because configuration is not valid, found the following errors:
- devices must be object
- groups must be object
```

**Reply by onedr0p** [reply]
@Koenkk do you have any guidance on making this work, or is it a bug?

**Reply by onedr0p** [reply]
There's a similuar issue with this one too:

```
ZIGBEE2MQTT_CONFIG_ADVANCED_LOG_OUTPUT="['console']"
# ZIGBEE2MQTT_CONFIG_ADVANCED_LOG_OUTPUT="[console]"
```

which yields:

```
Refusing to start because configuration is not valid, found the following errors:
- advanced/log_output must be array
```

**Reply by Koenkk** [reply]
Fixed the device/groups issue

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

> There's a similuar issue with this one too:

Use `ZIGBEE2MQTT_CONFIG_ADVANCED_LOG_OUTPUT='["console"]'`

**Reply by onedr0p** [reply]
I just updated to the latest version and confirm this is all working. This can be resolved thanks @Koenkk

**Reply by dmorel** [reply]
I'm failing to find any form that works for ZIGBEE2MQTT_CONFIG_ADVANCED_LOG_OUTPUT, `ZIGBEE2MQTT_CONFIG_ADVANCED_LOG_OUTPUT='["console"]'` returns the same error in my compose stack:
```
[zigbee2mqtt]   | Refusing to start because configuration is not valid, found the following errors:
[zigbee2mqtt]   | - advanced/log_output must be array
```
tried multiple forms, now i'm out of wits

**Reply by dmorel** [reply]
actually, this works with podman compose:
```
    environment:
      - ZIGBEE2MQTT_CONFIG_ADVANCED_LOG_LEVEL=warning
      - ZIGBEE2MQTT_CONFIG_ADVANCED_LOG_OUTPUT=["console"]
```
